plugin/evm/config/config.go

```diff
@@ -283,6 +289,32 @@ func (c *Config) SetDefaults() {
 	c.AcceptedCacheSize = defaultAcceptedCacheSize
 }
 
+// UpdateWithDefaults updates the configuration with defaults depending
+// on fields already set.
+func (c *Config) UpdateWithDefaults() {
```

Comment:
a bit out of the topic, but I think the best way to handle config defaults is to have a config with each field zero value being **not valid** (nil or `0` or `""` etc. depending on the use). Then it should go as:

1. read from config sources and load fields that are set into the config
2. set the defaults on the config for any field that is still the zero value of that field. That can include a bit of logic for fields depending on other fields (such as in this case below).

So for example you would never have a field with a `bool` type, it would always be a `*bool` and you would default it if it's still nil after reading it from a config source(s).